src/modules/print.js

```diff
@@ -115,54 +113,47 @@ export const showTasks = () => {
     listItemTitle.innerHTML = `${task.description}`;
     tasksDiv.appendChild(listItemTitle);
 
-    // Create a button to remove the book
     const removeButton = document.createElement('span');
     removeButton.setAttribute('class', 'close');
-    removeButton.getAttribute('id', 'remove-icon');
+    removeButton.setAttribute('id', 'remove-icon');
     removeButton.innerHTML = '🗑';
-    removeButton.addEventListener('click', (event) => {
-      const removeButtonParent = event.target.parentNode;
-      taskList.removeChild(removeButtonParent);
+    removeButton.addEventListener('click', () => {
+      taskList.removeChild(tasksDiv);
       taskList.removeChild(hr3);
+      removeTask(task.index);
+      showTasks();
     });
-
+    removeButton.style.display = task.completed ? 'inline' : 'none';
     tasksDiv.appendChild(removeButton);
-    inputCheckBox.addEventListener('change', (event) => {
-      handle.style.display = 'none';
-      const listItem = event.target.parentNode;
-      const removeButton = listItem.querySelector('.close');
-      removeButton.style.margin = '0 15px 0 0';
-      handle.style.display = event.target.checked ? 'none' : 'inline';
-      removeButton.style.display = event.target.checked ? 'inline' : 'none';
-    });
 
-    // Create the handle for dragging
     const handle = document.createElement('span');
     handle.setAttribute('class', 'handle');
     handle.innerHTML = '&#8942;';
     tasksDiv.appendChild(handle);
+    handle.style.display = task.completed ? 'none' : 'inline';
 
-    // Add event listener to the list item for editing
     listItemTitle.addEventListener('click', () => {
       listItemTitle.contentEditable = true;
       listItemTitle.focus();
     });
 
-    // Add event listener to handle changes in the edited list item
     listItemTitle.addEventListener('blur', () => {
       listItemTitle.contentEditable = false;
-      // Update the book title in the array
       tasks[i].description = listItemTitle.innerHTML;
-      saveTasks(); // Save the updated book list to local storage
+      saveTasks();
     });
 
-    // Append the books, author, and remove button to the book list
     taskList.appendChild(tasksDiv);
     const hr3 = document.createElement('hr');
     taskList.appendChild(hr3);
-  }
 
-  // Create "Clear All Completed" button
+    // Update the text decoration style based on the checkbox state
+    if (inputCheckBox.checked) {
+      listItemTitle.style.textDecoration = 'line-through';
+    } else {
+      listItemTitle.style.textDecoration = 'none';
+    }
+  }
   const clearCompletedButton = document.createElement('button');
   clearCompletedButton.setAttribute('class', 'clear-button');
   clearCompletedButton.textContent = 'Clear All Completed';
```

Comment:
- [ ] - [ ] Kindly consider implementing   Array.filter() on this ` clearCompletedButton ` to meet the below requirement
> Implement a function for the "Clear all completed" button (use filter() method).

Reply:
Filter method is used for "clear all completed" button
[
![image](https://github.com/Kaiserabbas/todo-list/assets/123939543/2d44f53c-a5c0-4899-81ed-360f839dcae9)
](url)